### ISSUE TITLE
--watch: Saving file causes 2 or 3 runs

### ISSUE THREAD

**Issue by lydell** [mpizenberg/elm-test-rs#61]
1. `elm-test-rs init`
2. `elm-test-rs --watch`
3. Edit `tests/Tests.elm` in an editor of choice: Add a space or something and save.

Result: The tests run 2 or 3 times.

Expected: The tests run once.

Note: You have to make a change in your editor. Just saving does not cause a re-run (not even `touch tests/Tests.elm`).

I did this on macOS.

**Comment by mpizenberg** [comment]
Thanks for the report! I'm not experiencing these issues on my system (archlinux) so I suspect this is platform dependent.

I'm using the [notify](https://github.com/notify-rs/notify) crate to watch files: https://github.com/mpizenberg/elm-test-rs/blob/e23ba0f7bc27f429d094770eca69f580d17e863e/src/make.rs#L55

I'm not very well versed with how to use it with best practices and there might be answers there. By the way, there is also a breaking version 5.0 in preparation for quite some time. It might solve the problem if there is no solution with the current stable version but I've no idea of how long it may take before they release.

In all cases I'd be very grateful if people with Rust knowledge and using macOS could look into this.

**Comment by mpizenberg** [comment]
Some debug info provided by @lydell showed that multiple triggers are cause by non elm-related files. Only reacting to elm related files will certainly help here.
```
Vim backupcopy=auto
Remove("/Users/lydell/stuff/elm-test-rs/tests/4913")
Create("/Users/lydell/stuff/elm-test-rs/tests/Tests.elm")
Remove("/Users/lydell/stuff/elm-test-rs/tests/Tests.elm~")

Vim backupcopy=yes
Write("/Users/lydell/stuff/elm-test-rs/tests/Tests.elm")
Remove("/Users/lydell/stuff/elm-test-rs/tests/Tests.elm~")

Vim backupcopy=no
Create("/Users/lydell/stuff/elm-test-rs/tests/Tests.elm")
Remove("/Users/lydell/stuff/elm-test-rs/tests/Tests.elm~")

IntelliJ
Write("/Users/lydell/stuff/elm-test-rs/tests/Tests.elm")
Remove("/Users/lydell/stuff/elm-test-rs/tests/Tests.elm~")

VSCode
Write("/Users/lydell/stuff/elm-test-rs/tests/Tests.elm")

echo '' >> tests/Tests.elm
Write("/Users/lydell/stuff/elm-test-rs/tests/Tests.elm")

touch tests/Foo.txt
Create("/Users/lydell/stuff/elm-test-rs/tests/Foo.txt")
```

**Comment by mpizenberg** [comment]
solved by #80
